### PR TITLE
Run tests in installed collection

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -78,29 +79,41 @@ dist-test: $(MANIFEST)
 	ANSIBLE_COLLECTIONS_PATHS=build/collections ansible-doc $(NAMESPACE).$(NAME).organization | grep -q "Manage Organization"
 
 $(MANIFEST): $(NAMESPACE)-$(NAME)-$(VERSION).tar.gz
+ifeq ($(COLLECTION_COMMAND),mazer)
+	# No idea, why this fails. But mazer is old and deprecated so unlikely to beeing fixed...
+	# mazer install --collections-path build/collections $<
+	-mkdir build/collections build/collections/ansible_collections build/collections/ansible_collections/theforeman build/collections/ansible_collections/theforeman/foreman
+	tar xf $< -C build/collections/ansible_collections/theforeman/foreman
+else
```
Comment (mdellweg) [inline Makefile:87]: Somehow, this seems to be enough to make ansible 2.8 happily execute the tasks for `filters` and `katello_manifest` all the modules requiring the `foreman_helper` fail.
Please anyone with a decent knowledge about "ansible 2.8 & mazer" speak up!

File: Makefile [hunk 2]
```diff
@@ -74,33 +75,44 @@ tests/test_playbooks/vars/server.yml:
...
 
-build:
+build/src:
 	-mkdir build build/src build/src/meta build/src/plugins $(addprefix build/src/plugins/,$(PLUGIN_TYPES))
 
-$(NAMESPACE)-$(NAME)-$(VERSION).tar.gz: $(addprefix build/src/,$(DEPENDENCIES)) | build
+$(NAMESPACE)-$(NAME)-$(VERSION).tar.gz: $(addprefix build/src/,$(DEPENDENCIES)) | build/src
+ifeq ($(COLLECTION_COMMAND),mazer)
+	mazer build build/src
+	cp releases/$@ .
```
Comment (evgeni) [inline Makefile:112]:
```suggestion
	mazer build --collection-path=build/src
	cp build/src/releases/$@ .
```

mazer is stupid

File: Makefile (hunk 1 above)
Comment (evgeni) [reply Makefile:86]: I'm not sure I like being put in the "decent mazer knowledge" camp, but I might have fixed it.

File: Makefile [hunk 3]
```diff
@@ -74,33 +75,44 @@ tests/test_playbooks/vars/server.yml:
 	@echo "For recording, please adjust $@ to match your reference server."
 
 dist-test: $(MANIFEST)
-	ANSIBLE_COLLECTIONS_PATHS=build/collections ansible -m $(NAMESPACE).$(NAME).organization -a "username=admin password=changeme server_url=https://foreman.example.test name=collectiontest" localhost | grep -q "Failed to connect to Foreman server"
```
Comment (evgeni) [inline Makefile:77]: why did you remove that?

File: Makefile [hunk 4]
```diff
@@ -74,33 +75,44 @@ tests/test_playbooks/vars/server.yml:
...
 	ANSIBLE_COLLECTIONS_PATHS=build/collections ansible-doc $(NAMESPACE).$(NAME).organization | grep -q "Manage Organization"
```
Comment (evgeni) [inline Makefile:78]:
```suggestion
	ANSIBLE_COLLECTIONS_PATHS=build/collections ansible -m $(NAMESPACE).$(NAME).organization -a "username=admin password=changeme server_url=https://foreman.example.test name=collectiontest" localhost | grep -q "Failed to connect to Foreman server"
	ANSIBLE_COLLECTIONS_PATHS=build/collections ansible-doc $(NAMESPACE).$(NAME).organization | grep -q "Manage Organization"
```

File: Makefile (hunk 3 above)
Comment (mdellweg) [reply Makefile]: That is a test, that our modules (pars pro toto) work as a collection.
We now test all the modules as a collection. So it felt duplicate.
We do not have anything similar for the docs (yet) however, and i left that.

File: Makefile (hunk 1 above)
Comment (mdellweg) [reply Makefile:87]: That is why i hesitated to ping you.

File: Makefile (hunk 3 above)
Comment (evgeni) [reply Makefile]: ah. I actually think that's still usefull, as it makes it easier to test "did the collection build fine" w/o running the testsuite.

**Comment by mdellweg** [comment]
Thanks for fixing it.
